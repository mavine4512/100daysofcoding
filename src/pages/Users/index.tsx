import React,{ useEffect,  useState } from 'react';
import { useDispatch,useSelector } from 'react-redux';
import {
    TableContainer,
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
} from '@material-ui/core';
import Paper from '@material-ui/core/Paper';
import useStyles from './style';
import { Users as IPeople } from '../../interface/Types/Users';
import { getUsers } from "../../features/actionCreator";
import { selectList } from "../../features/usersSlice";

const Users=():JSX.Element => {
    const { topHeaders,items,testLoading } = useStyles();
    const dispatch = useDispatch()
    const users = useSelector(selectList)

    const [loading,setLoading] = useState( true)

    useEffect(()=>{
        dispatch(getUsers())
        setLoading(false);

    },[dispatch])

    return (
        <div className="App" data-testid="table">
            <TableContainer data-testid="table" component={Paper}>
                <Table aria-label="simple table">
                    <TableHead >
                        <TableRow >
                            <TableCell className={topHeaders}>Name</TableCell>
                            <TableCell className={topHeaders} align="left">Address of city</TableCell>
                            <TableCell className={topHeaders} align="left">Address street</TableCell>
                            <TableCell className={topHeaders} align="left">Email</TableCell>
                            <TableCell className={topHeaders} align="left">Phone</TableCell>
                            <TableCell className={topHeaders} align="left">Company Name</TableCell>
                            <TableCell className={topHeaders} align="left">Website</TableCell>
                            <TableCell className={topHeaders} align="left">Phrase</TableCell>
                        </TableRow>
                    </TableHead>
                    { loading ? <p className={testLoading}> Loading... </p> :
                        <TableBody  data-testid="users">

                            {users.map((user:IPeople, index:number) => {
                                return(
                                    <TableRow key={index} >
                                        <TableCell className={items} align="left">{user.name}</TableCell>
                                        <TableCell className={items} align="left">{user.address.city}</TableCell>
                                        <TableCell className={items} align="left">{user.address.street}</TableCell>
                                        <TableCell className={items} align="left">{user.email}</TableCell>
                                        <TableCell className={items} align="left">{user.phone}</TableCell>
                                        <TableCell className={items} align="left">{user.company.name}</TableCell>
                                        <TableCell className={items} align="left">{user.website}</TableCell>
                                        <TableCell className={items} align="left">{user.company.catchPhrase}</TableCell>
                                    </TableRow >

                                )})}

                        </TableBody>}
                </Table>
            </TableContainer>
        </div>
    );
}

export default Users;
