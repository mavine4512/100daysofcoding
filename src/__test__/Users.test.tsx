import React from 'react';
import Users from "../pages/Users";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import store from "../App/store";

it('renders correctly users', () => {
    const tree = renderer.create(<Provider store={store}>
        <Users /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});



