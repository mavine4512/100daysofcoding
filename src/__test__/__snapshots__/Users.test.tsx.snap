// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
  data-testid="table"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="table"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Address of city
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Address street
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Phone
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Company Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Website
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Phrase
          </th>
        </tr>
      </thead>
      <p
        className="makeStyles-testLoading-5"
      >
         Loading... 
      </p>
    </table>
  </div>
</div>
`;

exports[`renders correctly users 1`] = `
<div
  className="App"
  data-testid="table"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="table"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Address of city
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Address street
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Email
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Phone
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Company Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Website
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topHeaders-1 MuiTableCell-alignLeft"
            scope="col"
          >
            Phrase
          </th>
        </tr>
      </thead>
      <p
        className="makeStyles-testLoading-5"
      >
         Loading... 
      </p>
    </table>
  </div>
</div>
`;
